### PR TITLE
[Driver][SYCL][FPGA] Do not generate host dependency files by default

### PR DESCRIPTION
**Pull request by mdtoguchi**
When compiling for -fsycl -fintelfpga, dependency file information needs
to be generated and added to the AOT compilation step.  The dep file is
only needed from the device compilation.

Remove the generation of the dependency files for the host compilation
unless the user explicitly calls for it (with a generation option)

Signed-off-by: Michael D Toguchi <michael.d.toguchi@intel.com>

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -1212,6 +1215,14 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-module-file-deps");
   }
 
+  if (!ArgM && Args.hasArg(options::OPT_fintelfpga) &&
```
Comment (meiyacha) [inline clang/lib/Driver/ToolChains/Clang.cpp:1218]: Would it be easier to understand the logic if we make this to an elsif instead? To me, it is, not sure about your view :).

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:1218]: Changes I made were based on other usage of `!ArgM` and the fact that the preceding `if` condition was quite large, reducing visibility of what the else would represent.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -1212,6 +1215,22 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
...
+    const char *DepFile;
+    createFPGATempDepFile(DepFile);
+    CmdArgs.push_back("-dependency-file");
+    CmdArgs.push_back(DepFile);
+    CmdArgs.push_back("-MT");
+    SmallString<128> P(Inputs[0].getBaseInput());
+    llvm::sys::path::replace_extension(P, "o");
+    SmallString<128> Quoted;
+    QuoteTarget(llvm::sys::path::filename(P), Quoted);
+    CmdArgs.push_back(Args.MakeArgString(Quoted));
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:1231]: Can we double-check these in the test?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:1231]: I'll adjust the test accordingly.  As for your `-foffload-static-lib` issue with `.d` question, this patch does not address that problem.  For that problem, we need to extract the .d files from the archive and add them to the aoc call.  Current device extraction only takes into account the device objects and passes them along to the partial link step.  The `.d` files are left alone.

Comment (meiyacha) [reply clang/lib/Driver/ToolChains/Clang.cpp:1231]: I suppose this also affect offload static library that are passed without this flag?

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:1231]: @meiyacha, that is correct.  Offload static lib w/ and w/out -foffload-static-lib behaves the same.

**Pull request merged**